test/test_runner/base_test_runner.go

```diff
@@ -57,10 +57,29 @@ func (t *BaseTestRunner) GetAgentRunDuration() time.Duration {
 func (t *TestRunner) Run(s ITestSuite) {
 	testName := t.TestRunner.GetTestName()
 	log.Printf("Running %v", testName)
-	testGroupResult, err := t.runAgent()
-	if err == nil {
+	/* 	This block is used to prevent userdata from running the agent
+		post-launch since it should've been done on launch already in
+		order to imitate customer behavoir
+	*/
+	testGroupResult := status.TestGroupResult{
+		Name: t.TestRunner.GetTestName(),
+		TestResults: []status.TestResult{
+			{
+				Name:   "Agent Started",
+				Status: status.SUCCESSFUL,
+			},
+		},
+	}
+	err := error(nil)
+	if testName == "Userdata" {
 		testGroupResult = t.TestRunner.Validate()
+	} else{
+		testGroupResult, err = t.runAgent()
```

Comment:
Take netstat_test as an example, its GetTestName() is defined as following
```
func (t *NetStatTestRunner) GetTestName() string {
	return "NetStat"
}
```
And since `Run` method used in base_test_runner has this line `testName := t.TestRunner.GetTestName()`, the if/else statement will go to else because the testName is NetStat

Reply:
Ok I understand now but I think for someone looking at this for the first time, its going to be extremely hard to correlate this if condition to the line [here](https://github.com/aws/amazon-cloudwatch-agent-test/pull/219/files#diff-21f80332df902cb88dd8d816f1fb3c758d8884f0a30b2054a44ec3c91044bb1bR39). Especially when these files are across folders (Ideally we should move this base_test_runner to some common place but restructuring this whole project probably needs to be a separate task).

I think what would be better is to add a method along the lines of `ShouldRunAgent()` in the base test runner that defaults to true and override this with a false just for the UserData implementation.